cni/pkg/install-cni/cmd/root.go

```diff
@@ -151,5 +153,12 @@ func constructConfig() (*config.Config, error) {
 		}
 	}
 
+	// Get install-cni DNS server addresses
+	dnsConfig, err := dns.ClientConfigFromFile("/etc/resolv.conf")
```

Comment:
I am not sure this is right. It should be from the pod, not CNI. They may not be the same... but I am not sure if we can do any better here either. 

Maybe parse `pod.spec.dnsConfig.nameservers`? seems flimsy though, I think the image can set them arbitrarily as well

Reply:
Are you suggesting parsing `dnsConfig.nameservers` first, and fall back to install-cni DNS server if not specified? Looking at the istio-cni log, I don't think dnsConfig is included in the pod spec as long as user does not specify it.

Reply:
I think `dnsConfig.nameservers` will override the defaults, so I think we would fallback to install-cni. But not 100% sure, or if there are other ways to modify /etc/resolv.conf (like modifying it in the image)

Reply:
Or make CNI DNS capture all request to 53. I know there is an issue with infinite loop due to local dns cache, but not sure if that is often an issue for k8s?

Reply:
Its probably extremely unlikely on k8s, it only happens when you have another local dns proxy